sc/src/main/java/uk/ac/ebi/atlas/experimentimport/ScxaExperimentDao.java

```diff
@@ -114,4 +117,13 @@ private String getPublicationIdsAsString(Collection<String> publicationIds) {
                         Collectors.joining(", "),
                         (result) -> result.isEmpty() ? null : result));
     }
+
+    public HashSet<String> getExperimentByGroupAccession(String experimentAccessionLike){
```

Comment:
You can change the return type here to be `Set<String>` even if you are returning a `HashSet`.

Reply:
Here I favour [Postel’s law](https://en.wikipedia.org/wiki/Robustness_principle) but I wonder if `HashSet` is the most appropriate type. If it’s a read-only `Set` I’d change it to `ImmutableSet`.